src/cmd/linuxkit/pkglib/pkglib.go

```diff
@@ -80,175 +80,189 @@ func NewFromCLI(fs *flag.FlagSet, args ...string) (Pkg, error) {
-	}
+		if hashPath == "" {
+			hashPath = pkgPath
+		} else {
+			hashPath, err = filepath.Abs(hashPath)
+			if err != nil {
+				return nil, err
+			}
 
-	dockerDepends, err := newDockerDepends(pkgPath, &pi)
-	if err != nil {
-		return Pkg{}, err
-	}
+			if !strings.HasPrefix(pkgPath, hashPath) {
+				return nil, fmt.Errorf("Hash path is not a prefix of the package path")
+			}
 
-	if devMode {
-		// If --org is also used then this will be overwritten
-		// by argOrg when we iterate over the provided options
-		// in the fs.Visit block below.
-		pi.Org = os.Getenv("USER")
-		if hash == "" {
-			hash = "dev"
+			// TODO(ijc) pkgPath and hashPath really ought to be in the same git tree too...
 		}
-	}
 
-	// Go's flag package provides no way to see if a flag was set
-	// apart from Visit which iterates over only those which were
-	// set.
-	fs.Visit(func(f *flag.Flag) {
-		switch f.Name {
-		case "disable-cache":
-			pi.DisableCache = *argDisableCache
-		case "enable-cache":
-			pi.DisableCache = !*argEnableCache
-		case "network":
-			pi.Network = *argNetwork
-		case "nonetwork":
-			pi.Network = !*argNoNetwork
-		case "org":
-			pi.Org = *argOrg
+		// make our own copy of piBase. We could use some deepcopy library, but it is just as easy to marshal/unmarshal
+		pib, err := yaml.Marshal(&piBase)
+		if err != nil {
+			return nil, err
+		}
+		var pi pkgInfo
+		if err := yaml.Unmarshal(pib, &pi); err != nil {
+			return nil, err
 		}
-	})
 
-	var srcHashes string
-	sources := []pkgSource{{src: pkgPath, dst: "/"}}
+		b, err := ioutil.ReadFile(filepath.Join(pkgPath, buildYML))
+		if err != nil {
+			return nil, err
+		}
 
-	for _, source := range pi.ExtraSources {
-		tmp := strings.Split(source, ":")
-		if len(tmp) != 2 {
-			return Pkg{}, fmt.Errorf("Bad source format in %s", source)
+		if err := yaml.Unmarshal(b, &pi); err != nil {
+			return nil, err
 		}
-		srcPath := filepath.Clean(tmp[0]) // Should work with windows paths
-		dstPath := path.Clean(tmp[1])     // 'path' here because this should be a Unix path
 
-		if !filepath.IsAbs(srcPath) {
-			srcPath = filepath.Join(pkgPath, srcPath)
+		if pi.Image == "" {
+			return nil, fmt.Errorf("Image field is required")
 		}
 
-		g, err := newGit(srcPath)
+		dockerDepends, err := newDockerDepends(pkgPath, &pi)
 		if err != nil {
-			return Pkg{}, err
-		}
-		if g == nil {
-			return Pkg{}, fmt.Errorf("Source %s not in a git repository", srcPath)
+			return nil, err
 		}
-		h, err := g.treeHash(srcPath, hashCommit)
-		if err != nil {
-			return Pkg{}, err
+
+		if devMode {
+			// If --org is also used then this will be overwritten
+			// by argOrg when we iterate over the provided options
+			// in the fs.Visit block below.
+			pi.Org = os.Getenv("USER")
+			if hash == "" {
+				hash = "dev"
+			}
 		}
 
-		srcHashes += h
-		sources = append(sources, pkgSource{src: srcPath, dst: dstPath})
-	}
+		// Go's flag package provides no way to see if a flag was set
```

Comment:
I do wonder whether it would be worthwhile using another flags library - maybe `spf13/viper`

Reply:
I think it probably would be, but I am unwilling to spend the time doing that surgery now.